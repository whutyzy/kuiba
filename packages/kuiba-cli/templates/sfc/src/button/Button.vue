<template>
  <button class="kuiba-button" >
    <slot />
  </button>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue'


export default defineComponent({
  name: 'KuibaButton',
  props: {},
  setup(props) {
    return {
      
    }
  }
})
</script>

<style lang="scss">
@import "./button.scss";
</style>
