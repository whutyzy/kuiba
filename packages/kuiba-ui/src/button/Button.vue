<template>
  <button class="kuiba-button" >
    <slot />
  </button>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue'


export default defineComponent({
  name: 'KuibaButton',
  props: {
    type: {type: String},
    size: {type: String},
    text: {type: String},
    color: {type: String},
    icon: {type: String},
    'icon-prefix': {type: String},
    'icon-position': {type: String},
    plain: {},
    round: {},
    disabled: {},
    loading: {} 

  },
  setup(props) {
    
    return {
      
    }
  }
})
</script>

<style lang="scss">
@import "./button.scss";
</style>
