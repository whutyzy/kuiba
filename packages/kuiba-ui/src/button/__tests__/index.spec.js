import example from '../example'
import Button from '..'
import KuibaButton from '../Button'
import { mount } from '@vue/test-utils'
import { createApp } from 'vue'

test('test button example', () => {
    const wrapper = mount(example)
    expect(wrapper.html()).toMatchSnapshot()
    wrapper.unmount()
})

test('test button plugin', () => {
    const app = createApp({}).use(Button)
    expect(app.component(Button.name)).toBeTruthy()
})

test('should emit click event', () => {
    const wrapper = mount(Button)
    wrapper.trigger('click')
    expect(wrapper.emitted().click).toHaveLength(1)
})

test('should not emit click event when disabled', () => {
    const wrapper = mount(Button, { props: { disabled: true } })
    wrapper.trigger('click')
    expect(wrapper.emitted('click')).toBeFalsy()
})

test('should not emit click event when loading', () => {
    const wrapper = mount(KuibaButton, { props: { loading: true } })
    wrapper.trigger('click')
    expect(wrapper.emitted('click')).toBeFalsy()
})

test('should render icon in the right side when setting icon-position to right', () => {
    const wrapper = mount(Button, {
        props: {
            icon: 'plus',
            iconPosition: 'right'
        }
    })
    expect(wrapper.html()).toMatchSnapshot()
})
