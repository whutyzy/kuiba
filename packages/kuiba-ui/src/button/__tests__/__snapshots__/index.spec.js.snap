// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button example 1`] = `
"<!-- <app-type>基本使用</app-type> -->
<!-- <basic-use/> -->
<!-- <app-type>修改颜色</app-type> --><button class=\\"basic-button\\" style=\\"background: rgb(3, 169, 244);\\">组件库从我开始</button>"
`;
